pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed failed

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed [failed](https://tanzanite.cluster.engflow.com/invocations/default/4c36439e-3a79-4fac-a462-a99246800e9c?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9tdWx0aXJlZ2lvbi05bm9kZS0zcmVnaW9uLTNhenMtdGVuYW50Om11bHRpcmVnaW9uLTlub2RlLTNyZWdpb24tM2F6cy10ZW5hbnRfdGVzdA==) on release-25.2 @ [0e37e8c11132a7175d402a2a03ecbcf2091d3781](https://github.com/cockroachdb/cockroach/commits/0e37e8c11132a7175d402a2a03ecbcf2091d3781):


```
=== RUN   TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed
I250411 05:27:18.637954 1 (gostd) rand.go:256  [-] 1  random seed: -7537059136602111478
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_multi_region_secondary_tenants_abstractions_allowed3967720823
    test_log_scope.go:76: use -show-logs to present logs inline
[05:27:24] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_secondary_tenants_abstractions_allowed with config multiregion-9node-3region-3azs-tenant: 18 tests, 1 failures
[05:27:25] --- total progress: 18 statements
--- total: 18 tests, 1 failures
    logic.go:4489: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_multi_region_secondary_tenants_abstractions_allowed3967720823
--- FAIL: TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed (7.00s)
=== RUN   TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed/privileged_range
    logic.go:2298: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_secondary_tenants_abstractions_allowed:83: 
        expected "only root users are allowed to modify the default range", but no error occurred
--- FAIL: TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed/privileged_range (0.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144275 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed failed [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_multi_region_secondary_tenants_abstractions_allowed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49307

--- a/pkg/ccl/logictestccl/testdata/logic_test/multi_region_secondary_tenants_abstractions_allowed
+++ b/pkg/ccl/logictestccl/testdata/logic_test/multi_region_secondary_tenants_abstractions_allowed
@@ -29,6 +29,12 @@ ALTER TENANT ALL SET CLUSTER SETTING sql.zone_configs.max_replicas_per_region = 
 
 user root
 
+# Wait for the setting to be propagated to the tenant cluster.
+query I retry
+SHOW CLUSTER SETTING sql.zone_configs.max_replicas_per_region
+----
+4
+
 # Our setting should not affect non-constraint related zone config updates.
 statement ok
 ALTER DATABASE db CONFIGURE ZONE USING gc.ttlseconds = 1
@@ -48,6 +54,12 @@ statement ok
 ALTER TENANT ALL SET CLUSTER SETTING sql.zone_configs.max_replicas_per_region = 3
 
 user root
+
+# Wait for the setting to be propagated to the tenant cluster.
+query I retry
+SHOW CLUSTER SETTING sql.zone_configs.max_replicas_per_region
+----
+3
 
 statement ok
 ALTER DATABASE db CONFIGURE ZONE USING gc.ttlseconds = 1
@@ -72,6 +84,12 @@ statement ok
 ALTER TENANT ALL SET CLUSTER SETTING sql.zone_configs.default_range_modifiable_by_non_root.enabled = false
 
 user root
+
+# Wait for the setting to be propagated to the tenant cluster.
+query B retry
+SHOW CLUSTER SETTING sql.zone_configs.default_range_modifiable_by_non_root.enabled
+----
+false
 
 statement ok
 ALTER RANGE default CONFIGURE ZONE USING num_replicas = 10

